src/gz/bufread.rs

```diff
@@ -34,7 +34,7 @@ fn read_le_u16<R: Read>(r: &mut R) -> io::Result<u16> {
     Ok((b[0] as u16) | ((b[1] as u16) << 8))
 }
 
-fn read_gz_header<R: Read>(r: &mut R) -> io::Result<GzHeader> {
+pub fn read_gz_header<R: Read>(r: &mut R) -> io::Result<GzHeader> {
```

Comment:
We could make this `pub(super)` so it's only public in the `gz` module and not potentially available to other crates.